src/common/common/obd_parser.cpp

```diff
@@ -6,51 +6,26 @@
 #include <iostream>
 #include <sstream>
 
-namespace {
-/**
-    Removes trailing/ leading whitespace from a string
-*/
-std::string cleanString(const std::string& line)
+std::unordered_map<std::string, std::string> parseObdData(const std::string& obdData)
 {
-    const char* WhiteSpace = " \t\v\r\n";
-    std::size_t start = line.find_first_not_of(WhiteSpace);
-    std::size_t end = line.find_last_not_of(WhiteSpace);
-    return start == end ? std::string() : line.substr(start, end - start + 1);
-}
-} // namespace
+	std::istringstream contents(obdData);
+	std::string key;
+	std::string value;
+	std::unordered_map<std::string, std::string> data;
 
-std::vector<DataBlock> getObdDataRaw(const std::string& obd)
-{
-    std::vector<DataBlock> data;
-    std::istringstream stream(obd);
-    std::string line;
+	while (contents >> key) {
+		contents >> value;
+		data.insert(std::make_pair(key, value));
```

Comment:
`std::unordered_map::emplace` can be used here

https://en.cppreference.com/w/cpp/container/unordered_map/emplace

This would shortern this code to `data.emplace(key, value);` as the pair would be constructed in place